### DISCUSSION TITLE
Auto off/shut for entities

### DISCUSSION THREAD

**Discussion by rlippmann** [home-assistant/architecture#1132] [Entity Models] [answered]
I was thinking it might be a good idea to add some sort of auto off/shut feature to certain entities (which would be optional).

So, when an entity state changes to on, you could have it automatically shut off after x minutes/seconds/whatever.

It could be useful for things like lights, fans, garage doors, switches in general, etc.

Currently you can do this with automations, but a more generic method might be useful so you don't need to write an automation for each entity that you want to have this functionality.

Thoughts?

**Comment by joostlek** [comment, marked as answer]
> but a more generic method might be useful so you don't need to write an automation for each entity that you want to have this functionality

Isn't that what we have blueprints for?

I think the current way with automations is the most flexible since it can work with any behaviour. You can't add customizations to the turn off, like "close the garage door after 5 minutes after being opened (which would work with this), unless my car is still detected in the doorway (which can't work with this)".

I think it's made for a specific use case and thus should belong in the automations and blueprints, since that can handle any edge cases.

**Reply by rlippmann** [reply]
Well, generally if I write an automation, I add a disable boolean select entity to disable the auto shut behavior, so that could be added to an auto shut entity (or mixin).

For your example, if your car is in the doorway, the photoelectric sensors won't let the door shut anyway, and there's no way to detect your car in the doorway anyway (since garage door openers tend to just control the open/shut part and have no feedback to the photoelectric sensors).

Personally, I've never bothered with blueprints.  It's easier for me to write the automation and tweak it myself.

**Reply by joostlek** [reply]
You can't add entities to entities.

Even if you want to implement such feature, I can already imagine 5 more of these features, and when you do that, you create a secondary way to automate, which undermines the current automations and only adds more technical depth, since you now need to support the same automation twice.

**Reply by gjohansson-ST** [reply]
Not to mention how this would conflict with automations and blueprints, user handling when they need something else and so on. 
I think what you're proposing is almost the concept of blueprints but it's automations and not stored by the entity itself.
Obviously not agreeing with this proposal.

**Reply by rlippmann** [reply]
> You can't add entities to entities.

I was thinking more of subclassing, i.e.:

```
class Light(AutoshutEntity)
```

> 
> Even if you want to implement such feature, I can already imagine 5 more of these features, and when you do that, you create a secondary way to automate, which undermines the current automations and only adds more technical depth, since you now need to support the same automation twice.

I see your point.
